apps/sms/views/conversation/js/conversation.js

```diff
@@ -2957,52 +2842,42 @@ var ConversationView = {
     new OptionMenu(params).show();
   },
 
-  discardDraft: function conv_discardDraft() {
-    // If we were tracking a draft properly, update the Drafts object entry.
-    if (!this.draft) {
-      return;
-    }
+  updateDraft() {
+    var thread = Threads.active;
 
-    Drafts.delete(this.draft).store();
+    // Need to assimilate recipients in order to check if any entered.
+    this.assimilateRecipients();
 
-    this.draft = null;
-  },
+    // If message content is not empty and/or we have _valid_ recipients in
+    // NewMessage view.
+    var hasDraftToSave = !Compose.isEmpty() ||
+      (!thread && this.recipients.numbers.length);
 
-  /**
-   * Saves currently unsent message content and/or recipients into a Draft
-   * object. Preserves currently active draft if specified. Draft preservation
-   * is intended to keep this.draft populated with the currently active draft
-   * when app is hidden, so that when app becomes visible again, it knows there
-   * is a draft to continue to keep track of.
-   *
-   * @param {boolean?} preserveDraft Optional boolean parameter that indicates
-   * whether or not we should preserve draft.
-   */
-  saveDraft: function conv_saveDraft(preserveDraft) {
-    var thread = Threads.active;
+    // If we have active draft, but user cleared it up, we should discard it.
+    if (this.draft && !hasDraftToSave) {
+      return Drafts.delete(this.draft).store().then(() => {
+        this.draft = null;
+      });
```

Comment:
wondering why we have to wait for store before setting it to null.
I'd do something like:
```javascript
Drafts.delete(this.draft);
this.draft = null;
return Drafts.store();
```